File: Module.Persist.TPM/Model/History/HistoricalPromoProductCorrectionPriceIncrease.cs
using Core.History;
using Module.Persist.TPM.Model.TPM;
using System;

namespace Module.Persist.TPM.Model.History
{
    [AssociatedWith(typeof(PromoProductCorrectionPriceIncrease))]
    public class HistoricalPromoProductCorrectionPriceIncrease : BaseHistoricalEntity<System.Guid>
    {

        public Guid PromoProductId { get; set; }
        public double? PlanProductUpliftPercentCorrected { get; set; }

        public Guid? UserId { get; set; }
        public string UserName { get; set; }

        public DateTimeOffset? CreateDate { get; set; }
        public DateTimeOffset? ChangeDate { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/promoproductcorrectionpriceincrease/HistoricalPromoProductCorrectionPriceIncrease.js
Ext.define('App.model.tpm.promoproductcorrectionpriceincrease.HistoricalPromoProductCorrectionPriceIncrease', {
    extend: 'Ext.data.Model',
    idProperty: '_Id',
    breezeEntityType: 'PromoProductCorrectionPriceIncrease',
    fields: [
        { name: '_Id', type: 'string', hidden: true },
        { name: '_ObjectId', hidden: true },
        { name: '_User', type: 'string', isDefault: true },
        { name: '_Role', type: 'string', isDefault: true },
        { name: '_EditDate', type: 'date', isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: '_Operation', type: 'string', isDefault: true },
        { name: 'PromoProductId', hidden: true, isDefault: true },
        { name: 'PlanProductUpliftPercentCorrected', type: 'float', hidden: false, isDefault: true },
        { name: 'UserId', hidden: true, isDefault: true, defaultValue: null },
        { name: 'UserName', type: 'string', hidden: false, isDefault: false },
        { name: 'CreateDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'ChangeDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'HistoricalPromoProductCorrectionPriceIncreases',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});
